src/methods/sep12/collectSep12Fields.ts

```diff
@@ -37,18 +38,47 @@ export const collectSep12Fields = async ({
 
   log.response({ title: "GET `/customer`", body: resultJson });
 
-  if (isNewCustomer && resultJson.status !== "NEEDS_INFO") {
+  if (isNewCustomer && resultJson.status !== Sep12CustomerStatus.NEEDS_INFO) {
     throw new Error(
       `Unexpected status for new customer \`${resultJson.status}\``,
     );
   }
 
+  const fieldsToCollect = Object.entries(resultJson.fields ?? {}).reduce(
+    (collectResult: any, field: any) => {
+      const [key, props] = field;
+
+      if (
+        !props.status ||
+        props.status === Sep12CustomerFieldStatus.NOT_PROVIDED ||
+        (props.status === Sep12CustomerFieldStatus.REJECTED &&
+          resultJson.status === Sep12CustomerStatus.NEEDS_INFO)
+      ) {
+        return { ...collectResult, [key]: props };
+      }
+
+      return collectResult;
+    },
+    {},
+  );
+
   if (resultJson.fields) {
     log.instruction({
-      title: "Received the following fields",
+      title: "Received the following customer fields",
       body: resultJson.fields,
     });
   }
 
-  return resultJson.fields;
+  if (Object.keys(fieldsToCollect).length > 0) {
```

Comment:
sooooo small - but this could be simplified to `(Object.keys(fieldsToCollect).length )`